modules/dxp/apps/portal-search-tuning/portal-search-tuning-gsearch-configuration-web/src/main/resources/META-INF/resources/js/components/Fragment.js

```diff
@@ -70,16 +67,40 @@ export default function Fragment({
 }) {
 	const [active, setActive] = useState(false);
 	const [expand, setExpand] = useState(true);
+	const [value, setValue] = useState(JSON.stringify(json, null, '\t'));
 
 	const editorElementRef = useRef();
 	const editorTextInputRef = useRef();
+	const valueRef = useRef(value);
+
+	useOutsideClick(editorTextInputRef);
 
 	useEffect(() => {
 		if (collapse) {
 			setExpand(false);
 		}
 	}, [collapse]);
 
+	function handleChange(newValue) {
+		valueRef.current = newValue;
+	}
+
+	function useOutsideClick(ref) {
+		useEffect(() => {
+			function handleClickOutside(event) {
+				if (ref.current && !ref.current.contains(event.target)) {
+					setValue(valueRef.current);
+				}
+			}
+
+			document.addEventListener('mousedown', handleClickOutside);
+
+			return () => {
+				document.removeEventListener('mousedown', handleClickOutside);
+			};
+		}, [ref]);
+	}
```

Comment:
Do we need this outside click? Shouldn't `handleChange` take care of storing the value?

Also instead of outside click we should use blur instead since there might not be an outside click if the user is using a keyboard to navigate.

Reply:
Oh I used outside click so it would know when to update the value state to what the current ref is... but I'm not sure if that's the best way.

I can try with blur!